### PR TITLE
Add machineconfig and machineconfigpool to enable nested-virtualization 

### PR DESCRIPTION
**Pull request by sherine-k**
Add machineconfig and machineconfigpool to enable nested-virtualization on nodes where kvm-device-plugin daemonset is deployed, so that the okd-team tekton pipeline for SCOS build runs faster.

### CODE REVIEW

**Comment by sherine-k** [comment]
cc @LorbusChris  @lmzuccarelli

File: cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/okd-team/clusterrolebinding.yaml
Comment (LorbusChris) [inline cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/okd-team/clusterrolebinding.yaml:13]: per the file path, this should be `okd-team`

File: cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/okd-team/clusterrolebinding.yaml
Comment (LorbusChris) [inline cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/okd-team/clusterrolebinding.yaml:5]: `-team` here as well

**Comment by LorbusChris** [comment]
I wonder whether this view could be scoped to Node and MC resources for `okd-team`.

Is there a known MC for enabling nested virt that we could also try to apply, e.g. per https://access.redhat.com/solutions/6692341?

**Review by durandom — COMMENTED** [review]
We try to avoid team specific bindings. It's better to create a generic `ClusterRoleBinding` and then add groups to it. Like https://github.com/operate-first/apps/blob/master/cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/cluster-logs-readers/clusterrolebinding.yaml
Although this one is not even deployed on e.g. smaug https://github.com/operate-first/apps/blob/master/cluster-scope/overlays/prod/moc/smaug/kustomization.yaml#L122-L135 🤷 
But I think you get the gist.

Currently nobody uses the `ClusterRole: view` on smaug: 

```
oc get clusterrolebinding --all-namespaces -o wide | grep "ClusterRole/view" | wc
       0       0       0
```

@tumido @HumairAK @larsks is there an existing role we could use for this purpose?

**Comment by durandom** [comment]
@sherine-k run `pre-commit` https://github.com/operate-first/apps/blob/master/contributing.md#linting to fix the pre-commit CI job

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml
Comment (LorbusChris) [inline cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml:11]:
```suggestion
  machineConfigSelector:
    matchExpressions:
      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,mcp-kvm-device-plugin]}
  nodeSelector:
    matchLabels:
      node-role.kubernetes.io/mcp-kvm-device-plugin: ""
```

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml
Comment (LorbusChris) [inline cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml:6]:
```suggestion
    machineconfiguration.openshift.io/role: "mcp-kvm-device-plugin"
```

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/daemonset.yaml
Comment (LorbusChris) [inline cluster-scope/base/apps/daemonsets/kvm-device-plugin/daemonset.yaml:50]:
```suggestion
        machineconfiguration.openshift.io/role: "mcp-kvm-device-plugin"
```

**Review by LorbusChris — CHANGES_REQUESTED** [review]
per https://access.redhat.com/solutions/5688941 I think we have to things a bit differently.
/hold

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/daemonset.yaml
Comment (sherine-k) [reply cluster-scope/base/apps/daemonsets/kvm-device-plugin/daemonset.yaml]: Done! great suggestion!

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml
Comment (sherine-k) [reply cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml]: Done

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml
Comment (sherine-k) [reply cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml]: Done! thx

**Comment by sherine-k** [comment]
@durandom @larsks this is now ready: We've taken away the cluster-scoped view permission for the moment. If this PR works, we probably wont need that. WDYT?
Would you mind taking a look

**Review by LorbusChris — APPROVED** [review]

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml
Comment (tumido) [inline cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml:1]: Please move these resources to a new folder in [`cluster-scope/base/machineconfiguration.openshift.io/machineconfigs`](https://github.com/operate-first/apps/tree/master/cluster-scope/base/machineconfiguration.openshift.io/machineconfigs). This new folder should be named `80-enable-nested-virt`. Resource of `machineconfiguration.openshift.io` api and `MachineConfig` kind should not be placed in `apps` api folder under `daemonsets` kind.

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml
Comment (tumido) [inline cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml:2]: Same as above, please create new folder in [`cluster-scope/base/machineconfiguration.openshift.io`](https://github.com/operate-first/apps/tree/master/cluster-scope/base/machineconfiguration.openshift.io) named `machineconfigpools/mcp-kvm-device-plugin` and place it there. Then either pull those resources individually into a `cluster-scope/overlays/<env>/<cluster>` (or create a bundle in `cluster-scope/bundles`).

Please read https://github.com/operate-first/apps/blob/master/cluster-scope/README.md (or 
https://github.com/operate-first/blueprint/blob/main/adr/0009-cluster-resources.md + https://github.com/operate-first/blueprint/blob/main/adr/0015-cluster-resources-amendment.md )

**Review by tumido — CHANGES_REQUESTED** [review]

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml
Comment (sherine-k) [reply cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfig.yaml]: Thanks @tumido , I'm new to this. I'll read up on the suggested docs.

File: cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml
Comment (sherine-k) [reply cluster-scope/base/apps/daemonsets/kvm-device-plugin/machineconfigpool.yaml]: Done as well

**Comment by durandom** [comment]
@Gregory-Pereira can you have a look into the prow issues?

**Comment by Gregory-Pereira** [comment]
I already am. I tested connectivity to the bucket yesterday which looks fine, seems like there is some issue with one of the three init containers that start with every prow job.

**Comment by LorbusChris** [comment]
Once this is merged, we'll need a cluster admin to label 3 nodes with the following in order to add them to the pool (can actually already be done beforehand too):
```
oc label node ip-192-168-130-218.example.internal node-role.kubernetes.io/mcp-kvm-device-plugin=
```

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
